worker/src/__tests__/evalService.test.ts

```diff
@@ -887,6 +893,960 @@ describe("eval service tests", () => {
     }, 10_000);
   });
 
+  describe("test eval filtering", () => {
+    const filterTest = test.extend<{
+      traceId1: string;
+      traceId2: string;
+      evalTemplate: EvalTemplate;
+      projectId: string;
+    }>({
+      projectId: "7a88fb47-b4e2-43b8-a06c-a5ce950dc53a",
+      traceId1: randomUUID(),
+      traceId2: randomUUID(),
+      evalTemplate: async ({}, use) => {
+        const evalTemplate = await prisma.evalTemplate.create({
+          data: {
+            id: randomUUID(),
+            projectId: "7a88fb47-b4e2-43b8-a06c-a5ce950dc53a",
+            name: "test-template",
+            version: 1,
+            prompt: "Evaluate this trace",
+            model: "gpt-3.5-turbo",
+            provider: "openai",
+            modelParams: { temperature: 0 },
+            vars: [],
+            outputSchema: {
+              type: "object",
+              properties: { score: { type: "number" } },
+            },
+          },
+        });
+
+        await use(evalTemplate);
+      },
+    });
+
+    filterTest(
```

Comment:
That's a pattern I haven't seen before. What's the idea here and why is it useful?

Could we cover this via a `it.each` instead to avoid some of the repetitive setups instead?